doc/specs/#888 - Com Api.md

```diff
@@ -0,0 +1,812 @@
+In this way, an app that takes over the com registration of the Windows Package Manager could use the user's mistake to allow itself to escalate.
+A proposed mitigation is provided in OpenCatalogAsync which is an optional call.
+
+Another question is whether it's required to provide a wrapper api for this api to avoid callers having to use CoCreateInstance, or write their own wrapper to project into c#.
+We could provide a 1 to 1 wrapper implementation in Microsoft.Management.Deployment.Client.
+
+Naming of the items is as always a tricky question. For this api there are multiple similar apis that are relevant with regard to naming and consistency. There is the
+Windows Package Manager command line which uses "source" to describe the various repositories that can host packages and "search" to describe looking up an app. 
+https://docs.microsoft.com/en-us/windows/package-manager/winget/
+There is the Windows::ApplicationModel::PackageCatalog which exists as a Windows API for installing packages and monitoring their installation progress.
+https://docs.microsoft.com/en-us/uwp/api/windows.applicationmodel.packagecatalog?view=winrt-19041
+And there is Windows.Management.Deployment.PackageManager which allows packages with the packageManagement capability to install msix apps and uses "Find" to describe looking up an app
+https://docs.microsoft.com/en-us/uwp/api/windows.management.deployment.packagemanager?view=winrt-19041 
+
+I've chosen to align with the Windows APIs in using *Catalog and Find. But this may cause some confusion if callers are using both the winget command line and this API.
+In particular a problem may be that since this API does not yet propose to implement adding a "source", client applications would need to work with the command line
+interface in order to do that which may cause the name change from Source to AppCatalog to be particularly confusing.
+
+# 5 API Details
+
+```c# (but really MIDL3)
+namespace Microsoft.Management.Deployment
+{
+    /// State of the install.
+    enum InstallProgressState
+    {
+        /// The install is queued but not yet active. Cancellation of the IAsyncOperationWithProgress in this state will prevent the app from downloading or installing.
+        Queued = 0,
+        /// The installer is downloading. Cancellation of the IAsyncOperationWithProgress in this state will end the download and prevent the app from installing.
+        Downloading = 1,
+        /// The install is in progress. Cancellation of the IAsyncOperationWithProgress in this state will not stop the installation or the post install cleanup.
+        Installing = 2,
+        /// The installer has completed and cleanup actions are in progress. Cancellation of the IAsyncOperationWithProgress in this state will not stop cleanup or roll back the install.
+        PostInstall = 3,
+    };

```

Comment:
There should be another one here called "Finished" (which might mean failed, but the progression has reached its end state.)